nixos/modules/services/web-apps/davis.nix

```
@@ -381,6 +381,7 @@ in
           APP_CACHE_DIR = "${cfg.dataDir}/var/cache";
           APP_LOG_DIR = "${cfg.dataDir}/var/log";
         };
+        phpPackage = lib.mkDefault cfg.package.passthru.php;
```

```suggestion
        phpPackage = lib.mkDefault cfg.package.passthru.php;
```
Looks better right?
But is still fishy to me. =/
Likely `phpPackage` should be overriden in the davis package.
I asked you, because you have better context. If it PHP version would never go out of sync from davis package, I think, davis package is the place to override the php version. So my previous suggestion was a bad one. Which I corrected in sequence. But give thought to this. I'm not invalidating what you are doing here. Just seeing if this really makes sense. I have no application context. :-)

Notice, if php is allowed to be of different versions, it would duplicate the version of php used. So I expect this to be the wrong place for this.

---

> Looks better right? 

Agreed, deferring to the package makes sense.

> Likely phpPackage should be overriden in the davis package.

`phpPackage` is an option on [services.phpfpm.pools.\<name>.phpPackage](https://search.nixos.org/options?channel=24.11&show=services.phpfpm.pools.%3Cname%3E.phpPackage&from=0&size=50&sort=relevance&type=packages&query=phpPackage+phpfpm), so the davis package cannot affect it. 

I don't know how much you know about PHP apps, but phpfpm is a small daemon responsible for executing the php scripts. The davis package itself doesn't have any binaries/executables, it's just a collection of php files and assets.

phpfpm creates a systemd service which is responsible for executing the php, basically boiling down to `php /nix/store/path/to/davis/index.php`

Due to upstream issues, it is crucial that (for now) davis is used with php83 and not php84. Hence the only real place to affect which version of php is used to execute davis is with this phpfpm `phpPackage` option.

Does that clarify the situation?

---

Humm... I still think this option is unnecessary because PHP package can be overridden at davis package and `services.phpfpm.pools.davis.phpPackage` in case you want PHP versions differently. But:

I'm not of a strong opinion on this. I was mostly checking if it made sense.
If you want to go ahead, I would not mind.